docs/docs/guides/migrate/sources/keycloak.md

```diff
@@ -0,0 +1,455 @@
+
+Next, select your service user that you created and select the **Org Owner** checkbox to assign the permissions of an organization owner to the service user.  
+
+<img src="/docs/img/guides/migrate/keycloak-43.png" alt="Migrating users from Keycloak to ZITADEL"/>
+
+### Generate an access token for the service user
+
+In order for the service user to access the API, they must be able to authenticate themselves. To authenticate the user, you can use either [JWT with Private Key](https://docs.zitadel.com/docs/guides/integrate/serviceusers#authenticating-a-service-user) flow (recommended for production) or [Personal Access Tokens](https://zitadel.com/docs/guides/integrate/pat)(PAT). In this tutorial we will choose the latter. 
+
+Go to **Users** -> **Service Users** again and click on the service user, then select **Personal Access Tokens** on the left and click the **+ New** button. Copy the generated personal access token to use it later.  Click **Close** after copying the PAT. 
+
+<img src="/docs/img/guides/migrate/keycloak-44.png" alt="Migrating users from Keycloak to ZITADEL"/>
+
+### Import user to ZITADEL via ZITADEL API
+
+if your Keycloak Realm has a single user, your `my-realm-users-0.json` file, into which you exported your Keycloak user previously, will look like this:
+
+```JSON
+{
+  "realm" : "my-realm",
+  "users" : [ {
+    "id" : "826731b2-bf17-4bd9-b45c-6a26c76ddaae",
+    "createdTimestamp" : 1693887631918,
+    "username" : "test-user",
+    "enabled" : true,
+    "totp" : false,
+    "emailVerified" : true,
+    "firstName" : "John",
+    "lastName" : "Doe",
+    "email" : "test-user@mail.com",
+    "credentials" : [ {
+      "id" : "c3f3759e-9d8a-4628-aad9-09e66f28a4e2",
+      "type" : "password",
+      "userLabel" : "My password",
+      "createdDate" : 1693888572700,
+      "secretData" : "{\"value\":\"ng6oDRung/pBLayd5ro7IU3mL/p86pg3WvQNQc+N1Eg=\",\"salt\":\"RaXjs4RiUKgJGkX6kp277w==\",\"additionalParameters\":{}}",
+      "credentialData" : "{\"hashIterations\":27500,\"algorithm\":\"pbkdf2-sha256\",\"additionalParameters\":{}}"
+    } ],
+    "disableableCredentialTypes" : [ ],
+    "requiredActions" : [ ],
+    "realmRoles" : [ "default-roles-my-realm" ],
+    "notBefore" : 0,
+    "groups" : [ ]
+  } ]
+}
+
+```
+
+Now, you need to transform the JSON to the ZITADEL data format by adhering to the ZITADEL API [specification](https://docs.zitadel.com/docs/apis/proto/management#importhumanuser) to import a user. The minimal format would be as shown below: 
+
+```JSON
+{
+    "userName": "test-user",
+    "email": {
+        "isEmailVerified": true,
+        "email":  "test-user@mail.com"
+    },
+    "profile": {
+        "firstName": "John",
+        "lastName": "Doe"
+    },
+    "hashedPassword": {
+        "value": "ng6oDRung/pBLayd5ro7IU3mL/p86pg3WvQNQc+N1Eg=",
+        "algorithm": "pbkdf2-sha256",
+        "salt": "RaXjs4RiUKgJGkX6kp277w==",
+        "hashIterations": "27500"
+    }
```

Comment:
As already discussed, the encoding has to be transformed in order for zitadel to understand it and being able to authenticate the user.

What I want to comment besides that, is that is also the incorrect json model, as we only except the `value` field in MCF.
https://zitadel.com/docs/apis/resources/mgmt/management-service-import-human-user

```
"hashedPassword": {
    "value": "$2y$12$hXUrnqdq1RIIYZ2HPytIIe5lXdIvbhqrTvdPsSF7o.jFh817Z6lwm"
  },
```

Reply:
Noted. Thanks.